aiida/restapi/translator/nodes/process/calculation/calcjob.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""
+Translator for calcjob node
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+from aiida.restapi.translator.nodes.process.process import ProcessTranslator
+
+
+class CalcJobTranslator(ProcessTranslator):
+    """
+    Translator relative to resource 'calculations' and aiida class Calculation
+    """
+
+    # A label associated to the present class (coincides with the resource name)
+    __label__ = 'calcjobs'
+    # The AiiDA class one-to-one associated to the present class
+    from aiida.orm import CalcJobNode
+    _aiida_class = CalcJobNode
+    # The string name of the AiiDA class
+    _aiida_type = 'process.calculation.calcjob.CalcJobNode'
+
+    _result_type = __label__
+
+    def __init__(self, **kwargs):
+        """
+        Initialise the parameters.
+        Create the basic query_help
+        """
+        # basic query_help object
+        super(CalcJobTranslator, self).__init__(Class=self.__class__, **kwargs)
+
+    @staticmethod
+    def get_input_files(node, filename):
+        """
+        Get the submitted input files for job calculation
+        :param node: aiida node
+        :return: the retrieved input files for job calculation
+        """
+        if node.node_type.startswith('process.calculation.calcjob.'):
```

Comment:
Is this check necessary? Shouldn't this only be called for a `CalcJobNode`? If the node is not a `CalcJobNode` why would the translator be an instance of `CalcJobTranslator`?